atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraLogHelper.java

```diff
@@ -68,4 +69,18 @@ private static String getUpperEndpoint(Range<LightweightOppToken> range) {
         }
         return range.upperEndpoint().toString();
     }
+
+    @Value.Immutable
+    interface HostAndIpAddress {
+        String host();
+
+        String ipAddress();
+
+        static HostAndIpAddress fromAddress(InetSocketAddress address) {
+            return ImmutableHostAndIpAddress.builder()
+                    .host(address.getHostString())
+                    .ipAddress(address.getAddress().getHostAddress())
```

Comment:
This will NPE if the address is unresolved.

Reply:
oops - the first time I did this in the RC I did realise this, but obviously forgot about it now!